🐛 [BUG] - Definevent macro does not support binary socket data

### Description

Compile error when a socket event is set up to receive binary data.

### Reproduction steps

```bash
1. Create a definevent statement that handles binary
2. ``cargo run`` or ``cargo build``
```


### Screenshots

_No response_

### Logs

_No response_

### OS

Linux

> > What's the error? could you send a code sample that triggers the error?
> 
> I can re-create it tonight, ill have to go back in my history and get the snippet

NVM, Found it:


```rs
fn on_connect(socket: SocketRef, Data(data): Data<Value>, players: Arc<Mutex<Vec<Player>>>) {
    // ... (previous code in on_connect remains the same)

    // Register events using the definevent macro
    define_event!(socket, 
        "test", events::test::main(),
    );

    let players_clone = Arc::clone(&players);
    define_event!(socket,
        "updatePlayerLocation", 
        move |s, d, b| update_player_location(s, d, b, players_clone.clone()),
    );

    let players_clone = Arc::clone(&players);
    define_event!(socket,
        "getOnlinePlayers", 
        move |s, d, b| get_online_players(s, d, b, players_clone.clone()),
    );

    let players_clone = Arc::clone(&players);
    define_event!(socket,
        "getPlayersWithLocations", 
        move |s, d, a, b| get_players_with_locations(s, d, a, b, players_clone.clone()),
    );

    let players_clone = Arc::clone(&players);
    define_event!(socket,
        "broadcastMessage", 
        move |d, b| broadcast_message(d, b, players_clone.clone()),
    );
}
```


Compile Output: 
```
error[E0277]: `Bin` doesn't implement `std::fmt::Debug`
  --> src/players.rs:10:15
   |
10 |         data, bin
   |               ^^^ `Bin` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
   |
   = help: the trait `std::fmt::Debug` is not implemented for `Bin`
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `Bin` is not an iterator
   --> src/players.rs:68:16
    |
68  |     socket.bin(bin).emit("messageBack", data).ok();
    |            --- ^^^ `Bin` is not an iterator
    |            |
    |            required by a bound introduced by this call
    |
    = help: the trait `Iterator` is not implemented for `Bin`, which is required by `Bin: IntoIterator`
    = note: required for `Bin` to implement `IntoIterator`
note: required by a bound in `socketioxide::socket::Socket::<A>::bin`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/socketioxide-0.13.1/src/socket.rs:596:36
    |
596 |     pub fn bin(&self, binary: impl IntoIterator<Item = impl Into<Bytes>>) -> ConfOperators<'_, A> {
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Socket::<A>::bin`

error[E0277]: `Bin` doesn't implement `std::fmt::Debug`
  --> src/players.rs:94:38
   |
94 |     println!("Binary payload: {:?}", bin);
   |                                      ^^^ `Bin` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
   |
   = help: the trait `std::fmt::Debug` is not implemented for `Bin`
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `{closure@src/macros.rs:24:31: 24:38}: MessageHandler<LocalAdapter, _>` is not satisfied
   --> src/macros.rs:24:31
    |
24  |                   app.on($path, move || { handler });
    |                       --        ^^^^^^^^^^^^^^^^^^^ the trait `MessageHandler<LocalAdapter, _>` is not implemented for closure `{closure@src/macros.rs:24:31: 24:38}`
    |                       |
    |                       required by a bound introduced by this call
    |
   ::: src/main.rs:112:5
    |
112 | /     define_event!(socket,
113 | |         "updatePlayerLocation", 
114 | |         move |s, d, b| update_player_location(s, d, b, players_clone.clone()),
115 | |     );
    | |_____- in this macro invocation
    |
note: required by a bound in `socketioxide::socket::Socket::<A>::on`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/socketioxide-0.13.1/src/socket.rs:228:12
    |
226 |     pub fn on<H, T>(&self, event: impl Into<Cow<'static, str>>, handler: H)
    |            -- required by a bound in this associated function
227 |     where
228 |         H: MessageHandler<A, T>,
    |            ^^^^^^^^^^^^^^^^^^^^ required by this bound in `Socket::<A>::on`
    = note: this error originates in the macro `define_event` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `{closure@src/macros.rs:24:31: 24:38}: MessageHandler<LocalAdapter, _>` is not satisfied
   --> src/macros.rs:24:31
    |
24  |                   app.on($path, move || { handler });
    |                       --        ^^^^^^^^^^^^^^^^^^^ the trait `MessageHandler<LocalAdapter, _>` is not implemented for closure `{closure@src/macros.rs:24:31: 24:38}`
    |                       |
    |                       required by a bound introduced by this call
    |
   ::: src/main.rs:118:5
    |
118 | /     define_event!(socket,
119 | |         "getOnlinePlayers", 
120 | |         move |s, d, b| get_online_players(s, d, b, players_clone.clone()),
121 | |     );
    | |_____- in this macro invocation
    |
note: required by a bound in `socketioxide::socket::Socket::<A>::on`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/socketioxide-0.13.1/src/socket.rs:228:12
    |
226 |     pub fn on<H, T>(&self, event: impl Into<Cow<'static, str>>, handler: H)
    |            -- required by a bound in this associated function
227 |     where
228 |         H: MessageHandler<A, T>,
    |            ^^^^^^^^^^^^^^^^^^^^ required by this bound in `Socket::<A>::on`
    = note: this error originates in the macro `define_event` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `{closure@src/macros.rs:24:31: 24:38}: MessageHandler<LocalAdapter, _>` is not satisfied
   --> src/macros.rs:24:31
    |
24  |                   app.on($path, move || { handler });
    |                       --        ^^^^^^^^^^^^^^^^^^^ the trait `MessageHandler<LocalAdapter, _>` is not implemented for closure `{closure@src/macros.rs:24:31: 24:38}`
    |                       |
    |                       required by a bound introduced by this call
    |
   ::: src/main.rs:124:5
    |
124 | /     define_event!(socket,
125 | |         "getPlayersWithLocations", 
126 | |         move |s, d, a, b| get_players_with_locations(s, d, a, b, players_clone.clone()),
127 | |     );
    | |_____- in this macro invocation
    |
note: required by a bound in `socketioxide::socket::Socket::<A>::on`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/socketioxide-0.13.1/src/socket.rs:228:12
    |
226 |     pub fn on<H, T>(&self, event: impl Into<Cow<'static, str>>, handler: H)
    |            -- required by a bound in this associated function
227 |     where
228 |         H: MessageHandler<A, T>,
    |            ^^^^^^^^^^^^^^^^^^^^ required by this bound in `Socket::<A>::on`
    = note: this error originates in the macro `define_event` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `{closure@src/macros.rs:24:31: 24:38}: MessageHandler<LocalAdapter, _>` is not satisfied
   --> src/macros.rs:24:31
    |
24  |                   app.on($path, move || { handler });
    |                       --        ^^^^^^^^^^^^^^^^^^^ the trait `MessageHandler<LocalAdapter, _>` is not implemented for closure `{closure@src/macros.rs:24:31: 24:38}`
    |                       |
    |                       required by a bound introduced by this call
    |
   ::: src/main.rs:130:5
    |
130 | /     define_event!(socket,
131 | |         "broadcastMessage", 
132 | |         move |d, b| broadcast_message(d, b, players_clone.clone()),
133 | |     );
    | |_____- in this macro invocation
    |
note: required by a bound in `socketioxide::socket::Socket::<A>::on`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/socketioxide-0.13.1/src/socket.rs:228:12
    |
226 |     pub fn on<H, T>(&self, event: impl Into<Cow<'static, str>>, handler: H)
    |            -- required by a bound in this associated function
227 |     where
228 |         H: MessageHandler<A, T>,
    |            ^^^^^^^^^^^^^^^^^^^^ required by this bound in `Socket::<A>::on`
    = note: this error originates in the macro `define_event` (in Nightly builds, run with -Z macro-backtrace for more info)
    ```

--- a/src/main.rs
+++ b/src/main.rs
@@ -100,8 +100,36 @@ fn on_connect(socket: SocketRef, Data(data): Data<Value>, players: Arc<Mutex<Vec
     
 
     // Register events using the definevent macro
-    define_event!(socket, 
-        "test", events::test::main(),
+    // N.B. due to issues that i haven't been able to track down, the define_event! macro may not
+    // work. however, please do try to use it first. if it breaks, then use socket.on
+    // i will fix the macro when i get back; coding on a phone isnt very ergonomic
+    // define_event!(socket, 
+    //     "test", events::test::main(),
+    // );
+
+    let players_clone = Arc::clone(&players);
+    socket.on("updatePlayerLocation", move |s, d|
+        update_player_location(s, d, players_clone.clone()),
+    );
+
+    let players_clone = Arc::clone(&players);
+    socket.on("getOnlinePlayers", move |s|
+        get_online_players(s, players_clone.clone()),
+    );
+
+    let players_clone = Arc::clone(&players);
+    socket.on("getPlayersWithLocations", move |s, d|
+        get_players_with_locations(s, d, players_clone.clone()),
+    );
+
+    let players_clone = Arc::clone(&players);
+    socket.on("getPlayersWithLocations", move |s, d| {
+        get_players_with_locations(s, d, players_clone.clone())
+    });
+
+    let players_clone = Arc::clone(&players);
+    socket.on("broadcastMessage", move |d|
+        broadcast_message(d, players_clone.clone()),
     );
 
     // Register events using the socketioxide API directly

